src/inject.js

```diff
@@ -55,6 +56,49 @@ const stringHandlers = {
             return val;
         }
     },
+
+    // With animationName we look for an object that contains keyframes and
+    // inject them as an `@keyframes` block, returning a uniquely generated
+    // name. The keyframes object should look like
+    //  animationName: {
+    //    from: {
+    //      left: 0,
+    //      top: 0,
+    //    },
+    //    '50%': {
+    //      left: 15,
+    //      top: 5,
+    //    },
+    //    to: {
+    //      left: 20,
+    //      top: 20,
+    //    }
+    //  }
+    // TODO(emily): figure out a way to call this `animation` instead of
+    // `animationName`.
+    animationName: (val) => {
+        if (typeof val === "object") {
+            // Generate a unique name based on the hash of the object. We can't
+            // just use the hash because the name can't start with a number.
+            // TODO(emily): this probably makes debugging hard, allow a custom
+            // name?
+            const name = `keyframe_${hashObject(val)}`;
+
+            // Since keyframes need 3 lays of nesting, we use `generateCSS` to
+            // build the inner layers and wrap it in `@keyframes` ourselves.
```

Comment:
It doesn't work, because aphrodite stringifies the values of the second layer, instead of generating more css. So, `a: { '0%': { left: 0 } }` generates `a { 0%: [object Object]; }` instead of `a { 0% { left: 0px; } }` like we want.